gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PermissionServiceImpl.java

```diff
@@ -102,4 +98,46 @@ public boolean hasManagementRights(String userId) {
         }
         return hasManagementRights;
     }
+
+    /**
+     * Checks whether the user has an appropriate management role, which means he has create/update/delete permission
+     * on Organization or Environment scope, with the exception of the Environment/Application scope (which does not
+     * allow the user to access management)
+     *
+     * @param user
+     * @return true if the user has an appropriate management role, else false
+     */
+    private boolean hasRelevantManagementRole(UserEntity user) {
+        if (user.getRoles() == null) {
+            return false ;
+        }
+        boolean hasOrganizationCreateUpdateDeleteRole = false;
+        boolean hasEnvironmentCreateUpdateDeleteRoleExceptApplication = false ;
+
+        for (UserRoleEntity userRoleEntity :  user.getRoles()) {
+            if (userRoleEntity.getPermissions() != null) {
+                RoleScope currentScope = userRoleEntity.getScope() ;
+                for (String permissionName : userRoleEntity.getPermissions().keySet()) {
+                    String permissionString = new String(userRoleEntity.getPermissions().get((permissionName)));
+                    boolean isCreateUpdateOrDelete = permissionString.contains("C") ||
+                            permissionString.contains("U") ||
+                            permissionString.contains("D");
+
+                    if (currentScope.equals(RoleScope.ORGANIZATION)
+                            && isCreateUpdateOrDelete) {
+                        hasOrganizationCreateUpdateDeleteRole = true;
```

Comment:
IMHO, this function should stop as soon as hasOrganizationCreateUpdateDeleteRole = true or hasEnvironmentCreateUpdateDeleteRoleExceptApplication = true.